server/src/main/java/org/apache/druid/server/initialization/jetty/CustomExceptionMapper.java

```diff
@@ -30,13 +31,20 @@
 @Provider
 public class CustomExceptionMapper implements ExceptionMapper<JsonMappingException>
 {
+  private static final Logger log = new Logger(CustomExceptionMapper.class);
+  public static final String ERROR_KEY = "error";
+  public static final String UNABLE_TO_PROCESS_ERROR = "unknown json mapping exception";
+
   @Override
   public Response toResponse(JsonMappingException exception)
   {
+    log.warn(exception.getMessage() == null ? UNABLE_TO_PROCESS_ERROR : exception.getMessage());
     return Response.status(Response.Status.BAD_REQUEST)
                    .entity(ImmutableMap.of(
-                       "error",
-                       exception.getMessage() == null ? "unknown json mapping exception" : exception.getMessage()
+                       ERROR_KEY,
+                       exception.getMessage() == null
+                       ? UNABLE_TO_PROCESS_ERROR
+                       : exception.getMessage().split(System.lineSeparator())[0]
```

Comment:
I donot think so that is the correct approach. We can eat up important bits of information which are helpful for debugging purposes. A well crafted error will suddenly look weird.

Reply:
Does a new error strategy work for you : https://github.com/apache/druid/pull/11711 as contributed here ?

Reply:
@cryptoe The issue here is the error stack trace that accompanies the error message. Refer to [comment](https://github.com/apache/druid/pull/16821#issuecomment-2285399506), which has been flagged as a security concern.

Just to clarify, we are still logging the complete exception details as-is in the logs for debugging purposes.

Regarding [11711](https://github.com/apache/druid/pull/11711), it applies exclusively to exceptions of type SanitizableException.